app/editor/src/components/wysiwyg/Wysiwyg.tsx

```diff
@@ -100,20 +104,41 @@ export const Wysiwyg = <T extends object>({
           .replace(/<br\s*\/?>/g, '[br]'),
         'text/html',
       );
-      doc.body.textContent =
+      const html =
         doc.body.textContent
           ?.replaceAll('[p]', '<p>')
           .replaceAll('[/p]', '</p>')
           .replaceAll('[br]', '<br>') || '';
-      setFieldValue(fieldName as string, doc.body.textContent);
-      setState({ ...state, html: doc.body.textContent });
+      if (html !== state.html) {
+        setState({ ...state, html: html });
+        setFieldValue(fieldName as string, html);
+      }
     }
   };
 
+  const extractTags = (values: string[]) => {
+    return tags
+      .filter((tag) =>
+        values.some((value: string) => value.trim().toLowerCase() === tag.code.toLowerCase()),
+      )
+      .map((tag) => tag);
+  };
+
+  const updateTags = (html: string) => {
+    const stringValue = html.match(/\[([^\]]+)\][^[\]]*$/s)?.pop();
```

Comment:
This won't handle a scenario where there is extra spaces after the square brackets.

Reply:
Currently it seems to be working as expected so we might not need to worry about extra spaces after the square brackets in this case, thanks